### ISSUE TITLE
The phone cannot load large 3D models

### ISSUE THREAD

**Issue by xx0618** [enable3d/enable3d#83]
I use phaserExtension.0.21, and there is no problem loading 3D models below 50mb. When loading models above 80mb, it will be very slow. If you use a mobile browser, you cannot open the 3D model. How should I optimize so that the phone can also display large 3D models?

**Comment by yandeu** [comment]
Please ask this question in the [three.js forum](https://discourse.threejs.org/).

**Comment by xx0618** [comment]
All right! |

**Comment by yandeu** [comment]
Did you find your answer? I guess it is a memory issue? Phones have less memory than a pc.

**Comment by xx0618** [comment]
Yes, the size of memory is one of the main reasons. A model that is easy to open on the PC side is very difficult on the mobile side.
Someone asked similar questions on the threejs forum. The answer is to optimize the 3D model. Two methods for optimizing the model are more feasible:
1. Use MESHLAB software to perform model reduction operations to optimize the 3D model. [](https://www.meshlab.net/)
2. Use obj-simplify to automatically optimize the 3D model. (Only models within 200mb can be optimized, if too large an error will be reported and the optimization cannot be completed) [](https://github.com/jonnenauha/obj-simplify)

(These two methods can only optimize model formats such as OBJ, but cannot directly optimize GLTF/GLB formats)

Under normal circumstances, I use MESHLAB to reduce the surface first, and then optimize it with obj-simplify. In this way, an OBJ model of about 300mb can be optimized to a GLB of about 60mb, and sometimes it will be smaller.

If you find a more effective way, please let me know!

**Comment by yandeu** [comment]
I know there is something called [Draco to compress gltf](https://threejs.org/docs/#examples/en/loaders/DRACOLoader), but I have never used it.

**Comment by xx0618** [comment]
I have used Draco. Although it can compress gltf/glb very well, it takes a long time to decompress. And some parts of the model will become transparent after compression. So I gave up.
There is also a [gltfpack](https://github.com/zeux/meshoptimizer/tree/master/gltf) method to compress gltf/glb, after compression, it is `EXT_meshopt_compression`. Because gltf/glb cannot be exported with GLTFExporter, it is not necessary.
